lib/src/recurrence_rule.dart

```diff
@@ -23,9 +23,9 @@ class RecurrenceRule {
     this.until,
     this.count,
     this.interval,
-    Set<int> bySeconds = const {},
-    Set<int> byMinutes = const {},
-    Set<int> byHours = const {},
+    List<int> bySeconds = const [],
+    List<int> byMinutes = const [],
+    List<int> byHours = const [],
```

Comment:
I previously converted these to a `SplayTreeSet` to ensure they have a consistent order when iterating over the elements ([`Set`](https://api.dart.dev/stable/3.0.6/dart-core/Set-class.html) doesn't guarantee this). This is no longer necessary with lists, so we can just store them directly:

```suggestion
    this.bySeconds = const [],
    this.byMinutes = const [],
    this.byHours = const [],
```

`bySeconds = List.of(bySeconds),` and Co. below can then be removed